syntax = "proto3";


service MessageEcho {
  // Methods

  rpc ServerEcho(ClientRequest) returns (ServerResponse) {}

  rpc ServerEchoStream(ClientRequest) returns (stream ServerResponse) {}

  rpc ServerEchoClientStream(stream ClientRequest) returns (ServerResponse) {}

  rpc ServerEchoStreamClientStream(stream ClientRequest) returns (stream ServerResponse) {}
}


message ClientRequest {
  string message = 1;
}


message ServerResponse {
  string message = 1;
}
